src/Nancy.Swagger.Annotations/SwaggerObjects/AnnotatedOperation.cs

```diff
@@ -44,33 +44,69 @@ public AnnotatedOperation(string name, MethodInfo handler, ISwaggerModelCatalog
                 .Select(CreateSwaggerResponseObject)
                 .ToDictionary(x => x.GetStatusCode().ToString(), y => (global::Swagger.ObjectModel.Response) y);
 
-            var infos = handler.GetParameters().Where(x => x.GetCustomAttributes<RouteParamAttribute>().Any()).ToList();
+            var paramsList = new List<Parameter>();
+            CreateSwaggerParametersFromMethodAttributes(handler, paramsList);
+            CreateSwaggerParametersFromParameters(handler, paramsList);
+            Parameters = paramsList;
 
-            Parameters = CreateSwaggerParameters(infos);
+        }
 
+        private void CreateSwaggerParametersFromMethodAttributes(MethodInfo info, List<Parameter> paramsList)
+        {
+            var routeParamsAttributes = info.GetCustomAttributes<RouteParamAttribute>(true);
+            foreach (var attrib in routeParamsAttributes)
+            {
+                if (string.IsNullOrEmpty(attrib.Name))
+                    throw new ArgumentNullException("Name", "RouteParam name cannot be null when used on method.");
+                if (attrib.ParamIn == ParameterIn.Body)
+                {
+                    if (paramsList.Where(x => x.GetType() == typeof(AnnotatedBodyParameter)).Any())
+                    {
+                        throw new ArgumentException("A method can only have one Body RouteParamAttribute");
+                    }
+
+                    if (attrib.ParamType == null)
+                    {
+                        throw new ArgumentNullException("ParamType", "ParamType for Body must be specified when RouteParam used on method.");
```

Comment:
You have mixed usage of RouteParamAttribute and RouteParam throughout the strings in this file. Maybe just pick one?